examples/torch/pearl_ml1_push.py

```diff
@@ -0,0 +1,163 @@
+    qf1 = ContinuousMLPQFunction(env_spec=augmented_env,
+                                 hidden_sizes=[net_size, net_size, net_size])
+
+    qf2 = ContinuousMLPQFunction(env_spec=augmented_env,
+                                 hidden_sizes=[net_size, net_size, net_size])
+
+    obs_space = akro.Box(low=-1, high=1, shape=(obs_dim, ), dtype=np.float32)
+    action_space = akro.Box(low=-1,
+                            high=1,
+                            shape=(params['latent_size'], ),
+                            dtype=np.float32)
+    vf_env = EnvSpec(obs_space, action_space)
+
+    vf = ContinuousMLPQFunction(env_spec=vf_env,
+                                hidden_sizes=[net_size, net_size, net_size])
+
+    policy = TanhGaussianMLPPolicy(env_spec=augmented_env,
+                                   hidden_sizes=[net_size, net_size, net_size])
+
+    context_conditioned_policy = ContextConditionedPolicy(
+        latent_dim=params['latent_size'],
+        context_encoder=context_encoder,
+        policy=policy,
+        use_information_bottleneck=params['use_information_bottleneck'],
+        use_next_obs=params['use_next_obs_in_context'],
+    )
+
+    pearl = PEARL(
+        env=env,
+        policy=context_conditioned_policy,
+        qf1=qf1,
+        qf2=qf2,
+        vf=vf,
+        num_train_tasks=params['num_train_tasks'],
+        num_test_tasks=params['num_test_tasks'],
+        latent_dim=params['latent_size'],
+        meta_batch_size=params['meta_batch_size'],
+        num_steps_per_epoch=params['num_steps_per_epoch'],
+        num_initial_steps=params['num_initial_steps'],
+        num_tasks_sample=params['num_tasks_sample'],
+        num_steps_prior=params['num_steps_prior'],
+        num_extra_rl_steps_posterior=params['num_extra_rl_steps_posterior'],
+        num_steps_per_eval=params['num_steps_per_eval'],
+        batch_size=params['batch_size'],
+        embedding_batch_size=params['embedding_batch_size'],
+        embedding_mini_batch_size=params['embedding_mini_batch_size'],
+        max_path_length=params['max_path_length'],
+        reward_scale=params['reward_scale'],
+    )
+
+    tu.set_gpu_mode(params['use_gpu'], gpu_id=0)
+    if params['use_gpu']:
+        pearl.to()
+
+    runner.setup(algo=pearl,
+                 env=env[0](),
+                 sampler_cls=LocalSampler,
+                 sampler_args=dict(max_path_length=params['max_path_length']),
+                 n_workers=1,
+                 worker_class=PEARLWorker)
+
+    worker_args = dict(deterministic=True, accum_context=True)
+    meta_evaluator = MetaEvaluator(runner,
+                                   test_task_sampler=test_env_sampler,
+                                   max_path_length=params['max_path_length'],
+                                   worker_class=PEARLWorker,
+                                   worker_args=worker_args,
+                                   n_test_tasks=params['num_test_tasks'])
+    pearl.evaluator = meta_evaluator
```

Comment:
is there no way to make this a constructor arg instead?

Reply:
I think no. Under the current API, there is a circular reference -- algo takes evaluator, evaluator takes runner and runner needs algo.policy in `runner.make_sampler`, which requires `runner.setup(algo, ...)` to be called.

Reply:
is there an issue open which would resolve this problem?

Reply:
Yes

Reply:
Okay, I didn't understand the real circular reference problem until yesterday, since I missed Anson's comment here.
I'll have it fixed soon, and then the `MetaEvaluator` can be constructed without the runner or algorithm.